src/Common/src/System/Security/Cryptography/EccKeyFormatHelper.cs

```diff
@@ -0,0 +1,434 @@
+                AsnEncodingRules.DER);
+
+            ret = new ECParameters
+            {
+                Curve = GetCurve(domainParameters),
+                Q =
+                {
+                    X = publicKeyBytes.Slice(1, fieldWidth).ToArray(),
+                    Y = publicKeyBytes.Slice(1 + fieldWidth).ToArray(),
+                },
+            };
+
+            ret.Validate();
+        }
+
+        private static void ValidateParameters(ECDomainParameters? keyParameters, in AlgorithmIdentifierAsn algId)
+        {
+            // At least one is required
+            if (keyParameters == null && algId.Parameters == null)
+            {
+                throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+            }
+
+            // If they are both specified they must match.
+            if (keyParameters != null && algId.Parameters != null)
+            {
+                ReadOnlySpan<byte> algIdParameters = algId.Parameters.Value.Span;
+                byte[] verify = ArrayPool<byte>.Shared.Rent(algIdParameters.Length);
+
+                try
+                {
+                    // X.509 SubjectPublicKeyInfo specifies DER encoding.
+                    // RFC 5915 specifies DER encoding for EC Private Keys.
+                    // So we can compare as DER.
+                    using (AsnWriter writer = AsnSerializer.Serialize(keyParameters.Value, AsnEncodingRules.DER))
+                    {
+                        if (!writer.TryEncode(verify, out int written) ||
+                            written != algIdParameters.Length ||
+                            !algIdParameters.SequenceEqual(verify.AsSpan(0, written)))
+                        {
+                            throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                        }
+                    }
+                }
+                finally
+                {
+                    // verify contains public information and does not need to be cleared.
+                    ArrayPool<byte>.Shared.Return(verify);
+                }
+            }
+        }
+
+        private static ECCurve GetCurve(in ECDomainParameters domainParameters)
+        {
+            if (domainParameters.Named == null)
+            {
+                throw new CryptographicException(SR.Cryptography_ECC_NamedCurvesOnly);
+            }
+
+            Oid curveOid = domainParameters.Named;
+
+            switch (curveOid.Value)
+            {
+                case Oids.secp256r1:
+                    curveOid = new Oid(Oids.secp256r1, nameof(ECCurve.NamedCurves.nistP256));
+                    break;
+                case Oids.secp384r1:
+                    curveOid = new Oid(Oids.secp384r1, nameof(ECCurve.NamedCurves.nistP384));
+                    break;
+                case Oids.secp521r1:
+                    curveOid = new Oid(Oids.secp521r1, nameof(ECCurve.NamedCurves.nistP521));
+                    break;
```

Comment:
The friendly names used here do not match the values in OidLookup, eg. `nistP256` vs. `ECDSA_P256`. This could result in problems when different named curves are compared in ECDsaCertificateExtensions.IsSameKey.

Reply:
The unit tests passed only because all of the implementations shared the same parsing code. I run the code on my second machine and accidentally recompiled only S.S.C.X509Certificates without also recompiling S.S.C.Algorithms. This combination made the unit tests fail because the implementation of ECCSecurityTransforms from different assemblies produced different results.

Reply:
I have reverted the change for now since it break Windows builds. However, I still think it could potentially be a problem and it definitely changes the returned values on macOS.